File: api/Entities/Finance/FinanceVoucher.cs
using System.ComponentModel.DataAnnotations;

namespace api.Entities.Finance
{
    public class FinanceVoucher: BaseEntity
    {
        public string PartyName { get; set; }
        [MaxLength(1)]
        public string Divn {get; set;}
        public int CoaId { get; set; }
        public string AccountName { get; set; }
        public int VoucherNo { get; set; }
        public DateTime VoucherDated {get; set;}
        public long Amount {get; set;}
        public string Narration {get; set;}
        [Required]
        public ICollection<VoucherEntry> VoucherEntries { get; set; }
        public ICollection<VoucherAttachment> VoucherAttachments { get; set; }  
        
        
    }
}

File: client/src/app/_models/finance/financeVoucher.ts
import { IVoucherAttachment } from "./voucherAttachment";
import { IVoucherEntry, VoucherEntry } from "./voucherEntry";

export interface IFinanceVoucher
{
	id: number;
	divn: string;
	coaId: number;
	accountName: string;
	voucherNo: number;
	voucherDated: Date;
	employeeId: number;
	reviewedById: number;
	reviewedByName: string;
	reviewedOn: Date;
	amount: number;
	narration: string;
	/*approved: boolean;
	approvedByEmployeeId: number;
	approvedOn: Date;
	*/
	voucherEntries: IVoucherEntry[];
	voucherAttachments: IVoucherAttachment[];
}

export class FinanceVoucher implements IFinanceVoucher
{
	id=0;
	divn = '';
	coaId = 0;
	accountName='';
	voucherNo = 0;
	voucherDated = new Date();
	employeeId = 0;
	reviewedById = 0;
	reviewedByName = '';
	reviewedOn=new Date();
	amount = 0;
	narration = '';
	/*approved= false;
	approvedByEmployeeId=0;
	approvedOn=new Date();
	*/
	voucherEntries: VoucherEntry[]=[];
	voucherAttachments: IVoucherAttachment[]=[];
}
